pkg/datarepair/checker/checker.go

```diff
@@ -120,12 +125,29 @@ func lookupResponsesToNodes(responses *pb.LookupResponses) []*pb.Node {
 	return nodes
 }
 
-// Run
+// Run the checker loop
 func (c *checker) Run() error {
-	return nil
-}
+	zap.S().Info("Checker is starting up")
+	ticker := time.NewTicker(c.interval)
+	defer ticker.Stop()
 
-// Stop
-func (c *checker) Stop() error {
-	return nil
+	go func() {
+		for range ticker.C {
+			zap.S().Info("Starting segment checker service")
+			go func() {
+				err := c.IdentifyInjuredSegments(c.ctx)
+				if err != nil {
+					c.errs = append(c.errs, err)
```

Comment:
This is possibly racy.

https://medium.com/@cep21/gos-append-is-not-always-thread-safe-a3034db7975